From 8ce8a1c2731d9208830ddd5b48a58980ee4ee9fe Mon Sep 17 00:00:00 2001
From: Xing Liu <xingliu@chromium.org>
Date: Tue, 18 Feb 2020 21:10:54 +0000
Subject: [PATCH 53/54] Download on Android Q: Remove a DCHECK in
 CreateReservation.

When resuming a download, the suggested path can be content URI. This
will hit a DCHECK in CreateReservation.

Bug: 1045642
Change-Id: I61c452afafbbc9606f747c84ff18e6b9dc9a0fb6
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2062759
Reviewed-by: Min Qin <qinmin@chromium.org>
Commit-Queue: Xing Liu <xingliu@chromium.org>
Cr-Commit-Position: refs/heads/master@{#742270}
---
 .../internal/common/download_path_reservation_tracker.cc        | 2 --
 1 file changed, 2 deletions(-)

diff --git a/components/download/internal/common/download_path_reservation_tracker.cc b/components/download/internal/common/download_path_reservation_tracker.cc
index f4bc7a4ee132..ea2032d721eb 100644
--- a/components/download/internal/common/download_path_reservation_tracker.cc
+++ b/components/download/internal/common/download_path_reservation_tracker.cc
@@ -326,8 +326,6 @@ PathValidationResult ValidatePathAndResolveConflicts(
 // - Returns the result of creating the path reservation.
 PathValidationResult CreateReservation(const CreateReservationInfo& info,
                                        base::FilePath* reserved_path) {
-  DCHECK(info.suggested_path.IsAbsolute());
-
   // Create a reservation map if one doesn't exist. It will be automatically
   // deleted when all the reservations are revoked.
   if (g_reservation_map == NULL)
-- 
2.25.2

